src/environment.rs

```diff
@@ -13,6 +13,16 @@ pub struct User {
     pub gid: nix::unistd::Gid,
 }
 
+impl User {
+    /// Get the user and group ids of the current user
+    fn current() -> Self {
+        let uid = nix::unistd::getuid();
+        let gid = nix::unistd::getgid();
+        debug!("Current user has uid {} and group {}", uid, gid);
```

Comment:
I wonder if this logging statement would be better placed in the `gather` function after `let user = ...`.